### PR TITLE
hw 8 (table)

### PR DESCRIPTION
**Pull request by ThisIsAntony**
- Add to html page table with header and some rows
- Add to header event listener to implement sorting functionality
  across the table data (alphabetical)
- Add to cells user interaction feature (click and edit text)

### CODE REVIEW

File: 8 table/script.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  nameCell.innerText = data.name;
+  nameCell.setAttribute("contenteditable", true);
+
+  const positionCell = document.createElement("td");
+  positionCell.classList.add("table-row-position");
+  positionCell.dataset.id = data.id;
+  positionCell.innerText = data.position;
+  positionCell.setAttribute("contenteditable", true);
+
+  row.append(numberCell, nameCell, positionCell);
```
Comment (kaseopea) [inline 8 table/script.js:21]: I would suggest to extract `append` method. `createRow` function responsibility would be only creating an html fragment. And than you can decide wheather to append it any DOM element.

File: 8 table/script.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  return row;
+};
+
+const clearTable = () => {
+  const allRows = document.querySelectorAll(".table-row");
+  allRows.forEach((item) => item.remove());
+};
+
+const fillTable = (data) => {
+  data.forEach((item, index) => table.append(createRow(item, index + 1)));
```
Comment (kaseopea) [inline 8 table/script.js:31]: `append` is expensive operation. I would suggest to create all markup before appending.

File: 8 table/script.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  positionCell.innerText = data.position;
+  positionCell.setAttribute("contenteditable", true);
+
+  row.append(numberCell, nameCell, positionCell);
+  return row;
+};
+
+const clearTable = () => {
+  const allRows = document.querySelectorAll(".table-row");
+  allRows.forEach((item) => item.remove());
```
Comment (kaseopea) [inline 8 table/script.js:27]: All ways to clear inner contents are described here:
https://stackoverflow.com/questions/3955229/remove-all-child-elements-of-a-dom-node-in-javascript

File: 8 table/script.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+const tableHeader = document.querySelector(".table-header");
+
+fillTable(data);
+const tableRows = document.querySelectorAll(".table-row-name");
+
+tableHeader.addEventListener("click", (e) => {
+  const element = e.target.closest("td");
+  if (!element) return;
+
+  if (element.classList.contains("table-header-name")) {
```
Comment (kaseopea) [inline 8 table/script.js:99]: Do not like this multiple if statement. You are using same `sorting` function. 
You just need to define different props.
For example, think about using data attributes

Comment (ThisIsAntony) [reply 8 table/script.js]: done.

File: 8 table/script.js (hunk 3 above)
Comment (ThisIsAntony) [reply 8 table/script.js]: done.

File: 8 table/script.js (hunk 2 above)
Comment (ThisIsAntony) [reply 8 table/script.js]: done.

**Pull request merged**